src/provider/provider.service.ts

```diff
@@ -1,4 +1,35 @@
-import { Injectable } from '@nestjs/common';
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { WalletType } from '@prisma/client';
+import { WalletService } from 'src/wallet/wallet.service';
 
 @Injectable()
-export class ProviderService {}
+export class ProviderService {
+    constructor(
+        private walletService: WalletService,
+    ) {}
+
+    async getProviderWallet(providerId: number) {
+        // get provider wallet
+        const providerWallet = await this.walletService.fetchWallet(providerId)
+        if(providerWallet == null) {
+            throw new HttpException("Wallet does not exist", HttpStatus.NOT_FOUND);
+        }
+
+        // check provider
+        if(providerWallet.type != WalletType.provider) {
+            throw new HttpException("Wallet does not belong to provider", HttpStatus.BAD_REQUEST);
+        }
+        return providerWallet;
+    }
+
+    async addCreditsToProvider(providerId: number, credits: number) {
+
+        // check provider
+        let providerWallet = await this.getProviderWallet(providerId)
```

Comment:
what is the use of providerWallet

Reply:
Provider wallet is for keeping track of accumulated credits which are due for payment from the admin